### PR TITLE
Feature #r115 add pagination for super admin page (main and universityList)

### PR DESCRIPTION
**Pull request by ostannja**
dev
## Click-Up

* [Main Click-Up ticket](https://app.clickup.com/t/m79xwj)

## Code reviewers

- [x] @Ihor-Kozachok
- [x] @TionCada
- [x] @PashaZ 

### Second Level Review

- [ ] @github_username

## Summary of issue

As a SuperAdmin i want to see pagination on main page so that I can see all university admins

## Summary of change

Paginations on SuperAdmin main page and UniversityList page were implemented. Pagination function component as a separate common component for SuperAdmin pages was outputed. Pagination component was tested.

## Testing approach

Unit tests check rendering of components and click fireevents

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/components/pagination/index.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    >
+      <div
+        id='prevPage'
+        data-testid='prevPage'
+        className={
+          currentPage === 1
+            ? `${styles.arrow} ${styles.arrow__prev} ${styles.arrowUnable}`
+            : `${styles.arrow} ${styles.arrow__prev}`
+        }
+        onClick={() => {
```
Comment (Igor-Kozachok) [inline src/components/pagination/index.tsx:28]: потрібен тест на клік

File: src/components/pagination/index.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      })}
+      <div
+        id='nextPage'
+        data-testid='nextPage'
+        className={
+          currentPage === totalPages
+            ? `${styles.arrow} ${styles.arrow__next} ${styles.arrowUnable}`
+            : `${styles.arrow} ${styles.arrow__next}`
+        }
+        onClick={() => {
```
Comment (Igor-Kozachok) [inline src/components/pagination/index.tsx:62]: Потрібен тест на клік

**Review by TionCada — CHANGES_REQUESTED** [review]
A lot of work has been done in this task so far. I would like to highlight that pagination was taken out in separate component, which is very good. But I have some advices which are worth paying attention to before merging the branch with dev:

1) If I'm right, these styles should not be applied in superAdminAccount component (src/components/superAdmin/superAdminAccount/superAdminAccount.module.scss), as they're already applied in the pagination component (src/components/pagination/pagination.module.scss) and they have no effect on layout.

2)  If I'm right, these styles should not be applied in universityListPage component (src/pages/superAdminPages/universityListPage/universityListPage.module.scss), as they're already applied in the pagination component (src/components/pagination/pagination.module.scss) and they have no effect on layout.

3) Please, make the container a little bit smaller on these two pages, so it don't occupy the full width of page. Also, add margin or padding right under the pagination component on both pages. 
1.

2.

**Comment by ostannja** [comment]
> A lot of work has been done in this task so far. I would like to highlight that pagination was taken out in separate component, which is very good. But I have some advices which are worth paying attention to before merging the branch with dev:
> 
> 1. If I'm right, these styles should not be applied in superAdminAccount component (src/components/superAdmin/superAdminAccount/superAdminAccount.module.scss), as they're already applied in the pagination component (src/components/pagination/pagination.module.scss) and they have no effect on layout.
> 2. If I'm right, these styles should not be applied in universityListPage component (src/pages/superAdminPages/universityListPage/universityListPage.module.scss), as they're already applied in the pagination component (src/components/pagination/pagination.module.scss) and they have no effect on layout.
> 3. Please, make the container a little bit smaller on these two pages, so it don't occupy the full width of page. Also, add margin or padding right under the pagination component on both pages.
> 
> 
> 
> 2.
>

**Review by TionCada — APPROVED** [review]

**Review by PashaZ — APPROVED** [review]

**Comment by PashaZ** [comment]
Well done

**Review by Igor-Kozachok — APPROVED** [review]

File: src/components/pagination/index.tsx (hunk 1 above)
Comment (ostannja) [reply src/components/pagination/index.tsx:28]: The rows of the code in pagination.test.js, which are responsible for those clicks are ##114 155 163. So if to comment that rows the coverage of test decrease and the result is on the picture. 
Знімок екрана 2021-06-23 121834
Cases are discribed if the currentPage is 2 or 3 with the number of total pages equal to 3

**Pull request merged**